update(reports): added monthly checklist as issue template (#76)
cal-itp/reports: .github/ISSUE_TEMPLATE/monthly_reports.md

# Checklist for monthly report generation

> ## Process should being on the first of the month and reports emailed out no later than 2 weeks later

Reviewer: typo, being -> begin

> ## Process should being on the first of the month and reports emailed out no later than 2 weeks later

Reviewer: also maybe, '2 weeks later' -> '2 weeks after that'

# Checklist for monthly report generation

## Process should being on the first of the month and reports emailed out no later than 2 weeks later

### Creating report data

``` python
pip install -r requirements
make generate_parameters
make all -j 30
```

- [ ] verifying that parameters are generated in subfolders
- [ ] verify in outputs that reports were generated
- [ ] verify that data subfolder is present

### Generate static site

```python  
npm run build
python -m http.server
```

- [ ] report month has been added to UI
- [ ] find example of agency with no validation errors (avalon), with validation notices (santa clara), and with daily metric errors (sacrt)

> ### Review

Reviewer 2: Either within or before review, there should be a checkbox for publishing to the development website.

> - [ ] Request review from Olivia

Reviewer 2: Just an idea to link things up using GitHub tagging.

```suggestion
- [ ] Request review of development site from Transit Data Quality Lead (@o-ram)
```

> 

Reviewer 2: Maybe we don't need to get this specific, but we could have separate tasks for requesting reviews/approvals and receiving reviews/approvals. Like as follows:

```suggestion
- [ ] Receive approval from Cal-ITP technical lead (@evansiroky) to deploy website to production
```

Thoughts?

### Deploy Reports

to push report data to the production bucket

```python  
make sync-prod
```

or to copy from the dev to prod.

```python
gsutil -m rsync -r -d gs://gtfs-data-test/report_gtfs_schedule/ gs://gtfs-data/report_gtfs_schedule/
```

If there are no changes between development and production rerun the last github action workflow run on main.

- [ ] verify in the github action that it is copying production report data

> 

Reviewer 2: Unneeded line break

Author: kept the line break to follow linter best practices

- [ ] verify that production site has been updated with most up to date month

### Email reports

#### Testing

> obtain test emails from evan. Verify with Olivia and calitp that email contents are correct. Update the config file

Reviewer 2: Just an idea to link things up with GitHub markup:

```suggestion
Obtain test emails from Cal-ITP technical lead (@evansiroky). Verify with Transti Data Quality Lead (@o-ram) and Cal-ITP comms lead that email contents are correct. Update the config file.
```

- [ ] update config file to have current month

> 

Reviewer 2: Suggestion to add a checkboxes as follows:

```suggestion
- [ ] Verify with Cal-ITP Technical Lead (@evansiroky) that test email list is correct
- [ ] Verify with Transit Data Quality Lead (@o-ram) that production email list is correct
```

> - [ ] verify from Olivia that email content is correct

Reviewer 2: The comms lead should be involved here too.

```suggestion
- [ ] Verify with Transit Data Quality Lead (@o-ram) and Comms Lead that email content is correct
```

##### Generate the template (if needed)

script execution in development

```python  
python 3_generate_report_emails.py development
```

> - [ ] send out emails to evan's test.csv using sandbox token.
> - [ ] send out emails to test.csv using production token

Reviewer 2: I believe these are in Google Sheets, so this could be noted in a more agnostic way as follows:

```suggestion
- [ ] send out emails to test email list using sandbox token.
- [ ] send out emails to test email list using production token
```

- [ ] verify that the emails pass visual inspection

#### Production

Once emails have passed visual inspection, change config file for production

```python  
python 3_generate_report_emails.py production 
```

pass prompt that asks if production is correct and verify email recipients are production
send out emails

> 

Reviewer 2: Can you add a checkbox for the sending of production emails?

```suggestion
- [ ] send out emails to production email list using production token
```
